api/opencl_runtime_layer.asciidoc

```diff
@@ -6374,6 +6409,20 @@ include::{generated}/api/version-notes/CL_MEM_D3D11_RESOURCE_KHR.asciidoc[]
       returns the _resource_ argument specified when _memobj_ was created.
 endif::cl_khr_d3d11_sharing[]
 
+ifdef::cl_ext_buffer_device_address[]
+| {CL_MEM_DEVICE_ADDRESS_EXT_anchor}
+
+include::{generated}/api/version-notes/CL_MEM_DEVICE_ADDRESS_EXT.asciidoc[]
+  |  {cl_mem_device_address_EXT_TYPE}[]
+    | If _memobj_ was created using {clCreateBufferWithProperties} with
+      the {CL_MEM_DEVICE_PRIVATE_ADDRESS_EXT} property set to {CL_TRUE},
+      returns a list of device addresses for the buffer, one for each
+      device in the context in the same order as the list of devices
+      passed to {clCreateContext}.
```

Comment:
Now that `CL_MEM_DEVICE_PRIVATE_ADDRESS_EXT` can be passed if some devices don't support the extension, will this query return `NULL` or an undefined value for devices not supporting it? Or would one have to filter those out leaving no gaps?

Reply:
I think it has to be "undefined" because NULL might be a valid value in the device, unless we define it to be not.

Reply:
`NULL` doesn't imply it being 0, just that it's a NULL pointer, so runtimes could return whatever value represents `NULL` for that device.

Though I don't mind it being undefined either.

Reply:
Yes, but then we'd need to have an API for figuring out what is the "NULL" for each device.. I think it might be simplest and clearest to leave it undefined (likely 0).

Reply:
yeah, undefined is fine. One could argue that `NULL` issue exists with `clSetKernelArgDevicePointerEXT`, but one can use `clSetKernelArg` to set a `NULL` pointer, so yeah, probably better not to get into the `NULL` topic if we can avoid it.

Reply:
Yes and I even expanded the semantics to allow invalid pointers of any kind (like C/C++ calls do), as long as they are not dereferenced.